### PR TITLE
fix(tools): report tool success/failure via ToolResultBlock state

### LINKED ISSUE #2745: [Bug]: FileSystemTool 没有正确返回ToolResultState.ERROR
**<u>AgentScope-Java is an open-source project. To involve a broader community, we recommend asking your questions in English.</u>**

**Describe the bug**
FileSystemTool调用失败，例如write_file，返回字符串"Error: " + r.error()，不符合ReActAgent.determineToolResultState的判断，返回的仍然是ToolResultState.SUCCESS

[FilesystemTool.java (line 88)]
    @Tool(
            name = "write_file",
            description = "Write content to a new file, creating parent directories if needed.")
    public String writeFile(
            RuntimeContext runtimeContext,
            @ToolParam(name = "path", description = "Target file path") String path,
            @ToolParam(name = "content", description = "File content to write") String content) {
        WriteResult r = abstractFilesystem.write(runtimeContext, norm(path), content);
        return r.isSuccess() ? "Written to " + r.path() : "Error: " + r.error();
    }

[ReActAgent.java (line 2883)]
        private ToolResultState determineToolResultState(ToolResultBlock result) {
            if (result.isSuspended()) {
                return ToolResultState.RUNNING;
            }
            if (result.getState() != null && result.getState() != ToolResultState.RUNNING) {
                return result.getState();
            }
            if (result.getOutput() != null
                    && result.getOutput().stream()
                            .anyMatch(
                                    b ->
                                            b instanceof TextBlock tb
                                                    && tb.getText() != null
                                                    && tb.getText().startsWith("[ERROR]"))) {
                return ToolResultState.ERROR;
            }
            return ToolResultState.SUCCESS;
        

**Expected behavior**
WriteResult r.isSuccess() == false 时，返回ToolResultState.ERROR

### PR DESCRIPTION
**Pull request by haosong384**
## Problem

Built-in tools can return an error message while their calls are reported as `SUCCESS` (#2745). Inferring failure from text is also unsafe: reading a file or receiving a subagent reply containing `Error:` or `[ERROR]` can be a successful operation.

## Changes

- Add `ToolResultBlock.success(String)` to declare success explicitly. Unlike `text(...)`, it does not leave the result in the default `RUNNING` state.
- Return structured results from 42 annotated tool methods (plus the two filesystem search convenience overloads):

  | Tools | Methods |
  |---|---:|
  | FilesystemTool | 6 |
  | SessionSearchTool / TaskTool | 3 each |
  | WebTools | 2 |
  | MemoryGetTool / MemorySaveTool / ShellExecuteTool | 1 each |
  | TeamTool (dispatcher and 18 aliases) | 19 |
  | AgentSpawnTool (`agentSpawn`, `agentSend`, `agentList`) | 3 |
  | AgentGenerateTool / MemorySearchTool / TodoTools | 1 each |

- Preserve team JSON and subagent result payloads while assigning explicit states. Background acceptance and timeout promotion are `SUCCESS`; synchronous execution failures and forced synchronous timeouts are `ERROR`; cancelled or interrupted remote waits and interrupted `task_output` waits are `INTERRUPTED`.
- Treat an empty memory search query as `ERROR`; a search with no matches remains `SUCCESS`.
- Leave `ReActAgent` unchanged. Its existing fallback remains available for tools without an explicit terminal state.
- Add regression coverage for structured states, error-looking successful content, team aliases and failures, subagent execution and remote cancellation, generation/write errors, and todo validation. Update direct test callers and English/Chinese tool documentation.
- Integrate `main` at `3a9f9ec0`, retaining bounded filesystem search/listing output, task tracking status messages, and web interrupt handling. Add regression tests for the merged task error and interruption paths.

## Compatibility

These public Java methods, including the filesystem search convenience overloads, change from `String` to `ToolResultBlock`, or from `Mono<String>` to `Mono<ToolResultBlock>` for `agentSpawn`, `agentSend`, and `agentGenerate`. Direct callers must migrate to `getState()` and the text blocks in `getOutput()`; reflective `Toolkit` registration is unchanged.

Structured text reaches the model without the JSON string quoting previously applied to plain String returns. Task lookup errors retain the main branch's `invalid_request` and `not_found` status messages. An empty memory search query returns `Error: No query provided`. Non-zero shell exit codes retain their existing interpretation as successful tool execution with the exit code in the output.

## Validation

```text
Linux / JDK 17, after merging main:
mvn -B -pl agentscope-core,agentscope-harness -am clean test

agentscope-core: 2339 tests, 0 failures, 0 errors, 9 skips
agentscope-harness: 1095 tests, 0 failures, 0 errors, 9 skips
BUILD SUCCESS
```

No additional tests were excluded from the Linux run. Spotless checks pass for both modules. Windows compilation also passed after resolving the merge conflicts. The isolated Linux validation copy was checked against all 1,029 relevant source/resource/POM files in the submitted workspace.

Closes #2745

### CODE REVIEW

File: agentscope-core/src/main/java/io/agentscope/core/ReActAgent.java [hunk 1]
```diff
@@ -3195,12 +3195,32 @@ private ToolResultState determineToolResultState(ToolResultBlock result) {
...
             return ToolResultState.SUCCESS;
         }
 
+        /**
+         * Heuristic: checks whether the text signals an error result.
+         *
+         * <p>Recognised prefixes are {@code [ERROR]} (used by {@link ToolResultBlock#error(String,
+         * String)}) and {@code Error: } (used by all built-in {@code @Tool} string-returning
+         * methods). When the text is a JSON-quoted string (produced by
+         * {@link io.agentscope.core.tool.DefaultToolResultConverter} for String returns), the
```
Comment (AgentScopeJavaBot) [inline agentscope-core/src/main/java/io/agentscope/core/ReActAgent.java]: [nit] Consider using `Character.toString('"')` or a constant for the quote character to improve readability, though the current `char` comparison is idiomatic and correct. Minor style nit only.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This PR fixes issue #2745 where `@Tool` methods returning error strings with `"Error: "` prefix were misclassified as `SUCCESS` instead of `ERROR`. The root cause was twofold: `determineToolResultState` only checked for `[ERROR]` prefix, and `DefaultToolResultConverter` wraps strings in JSON quotes, preventing even that check from working. The fix extracts a `looksLikeError(String)` helper that strips one layer of JSON quotes and recognizes both prefixes. The implementation is minimal, well-documented, and includes comprehensive tests (both E2E and boundary cases). The PR description thoroughly explains the design trade-offs and downstream impact analysis.

File: agentscope-core/src/main/java/io/agentscope/core/ReActAgent.java (hunk 1 above)
Comment (AgentScopeJavaBot) [inline agentscope-core/src/main/java/io/agentscope/core/ReActAgent.java]: [nit] Consider using `Character.toString('"')` or a constant for the quote character to improve readability, though the current `char` comparison is idiomatic and correct. Minor style nit only.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This PR fixes issue #2745 where `@Tool` methods returning error strings with `"Error: "` prefix were misclassified as `SUCCESS` instead of `ERROR`. The root cause was twofold: `determineToolResultState` only checked for `[ERROR]` prefix, and `DefaultToolResultConverter` wraps strings in JSON quotes, preventing even that check from working. The fix extracts a `looksLikeError(String)` helper that strips one layer of JSON quotes and recognizes both prefixes. The implementation is minimal, well-documented, and includes comprehensive tests (both E2E and boundary cases). The PR description thoroughly explains the design trade-offs and downstream impact analysis.

**Comment by jujn** [comment]
我更倾向于重构一下：让 FilesystemTool 等工具成功/失败时返回结构化 ToolResultBlock.withState()；
目前，ReActAgent # determineToolResultState 取 ToolResultBlock 携带的 TextBlock.getText，容易误判：内容返回型工具可能会误报 ERROR

**Comment by haosong384** [comment]
@jujn Thanks — you were right, and the point turned out to be sharper than a style preference. Reworked as requested.

**Your concern reproduces.** `FilesystemTool.readFile` returns raw file content on success, so reading a log file whose first line is `Error: connection refused` is a successful read that my previous revision classified as ERROR:

```
expected: <SUCCESS> but was: <ERROR>
```

Five tools are exposed this way — `read_file`, `web_fetch`, `web_search`, `memory_get`, `shell execute` — all of them returning external content on success. Widening the prefix match traded a missed failure for a fabricated one, so it's gone.

**What changed**

- `ReActAgent` is now byte-identical to `main`; `git diff main -- ReActAgent.java` is empty. The fix lives entirely in the tools.
- 17 `@Tool` methods across `FilesystemTool`, `WebTools`, `SessionSearchTool`, `TaskTool`, `MemoryGetTool`, `MemorySaveTool`, `ShellExecuteTool` return `ToolResultBlock`: failures via `ToolResultBlock.error(msg)`, successes via a new `ToolResultBlock.success(text)`.
- Error text is unchanged — `error(msg)` already renders `"Error: " + msg`.

**Two findings worth your attention**

*`withState(SUCCESS)` is required on success paths, not optional.* An unset state is coerced to `RUNNING` by the constructor, and `determineToolResultState` treats `RUNNING` as "unknown" and falls through to the text fallback. So `ToolResultBlock.text(...)` does **not** opt a tool out of the heuristic. That's why I added `success(String)` — symmetric with the existing `error(String)`, and documented so the next person doesn't hit the same trap.

**One behaviour change to flag:** successful output loses the JSON quotes that `DefaultToolResultConverter` added around String returns, because `ToolResultBlock` passes through unserialised. The model now sees bare content and real newlines instead of `"\"...\""` with `\n` escapes. Looks like a strict improvement, but it does alter prompt content.

**Scope question for you.** This PR covers the tool named in #2745 plus every content-returning tool. Still returning strings: `TeamTool` (20 methods), `AgentSpawnTool` / `AgentGenerateTool` (`Mono<String>` → `Mono<ToolResultBlock>`), `TodoTools`, `MemorySearchTool`. I stopped here to keep the diff reviewable — want me to extend this PR, or follow up separately?

Verification: `mvn -pl agentscope-core,agentscope-harness clean test` → 2277 + 833 green, spotless clean. Note `clean` matters: incremental compilation hid a real return-type mismatch in `TaskTool` while still reporting BUILD SUCCESS.

**Comment by jujn** [comment]
👍 Please continue in the current PR.

**Comment by oss-maintainer** [comment]
⚠️ **Merge conflict detected**

This PR currently conflicts with `main` and cannot be merged. Please rebase (or merge `main` in) and resolve the conflicts:

```bash
git fetch origin
git rebase origin/main
# resolve conflicts, then:
git push --force-with-lease
```

This is a one-time reminder. Feel free to @mention me for a re-review after conflicts are resolved.

---
*Automated notification by github-manager-bot*
